core/execution/special_orders.go

```diff
@@ -261,10 +269,14 @@ func (m *Market) updateLPOrders(
 	for _, order := range allOrders {
 		order.UpdatedAt = now
 
+		_, toCancel := cancelIDs[order.ID]
+		_, toSubmit := submitIDs[order.ID]
 		// these order were actually cancelled, just send the event
-		if _, ok := cancelIDs[order.ID]; ok {
-			order.Status = types.OrderStatusCancelled
-			orderEvts = append(orderEvts, events.NewOrderEvent(ctx, order))
+		if toCancel {
+			if !toSubmit {
```

Comment:
something is not clear to me though is not a regression from the previous behaviour. Suppose an order id is in the submit and is not in the cancel - wouldn't it be submitted here and then again submitted in line 286?

Reply:
Nope, I asked myself the same thing, and basically here the allOrders is all that were in the book? The cancels are all that needs to cancel, and the submit are all that are new order instructed to be created (as in they were not in the book before but they are now needed)

Reply:
so the order of things as I understand it is the following:

1. we remove all LP orders from the book
`m.removeLPOrdersFromBook(ctx, lpOrders)`

2. then we resubmit all pegged orders 
	
```
partiesPos := map[string]events.MarketPosition{}
if needsPeggedUpdates {
_, partiesPos = m.reSubmitPeggedOrders(ctx, toSubmit)
}
```	

3. then we're calling update to reprice LP order
`newOrders, cancels := m.liquidity.Update(ctx, minLpPrice, maxLpPrice, m.repriceLiquidityOrder)`

and this newOrders and cancels is what goes into the function you changed above. So I can see that submits are not in the book as they've been removed. I'm not sure how there can be cancels if all orders have been removed in step 1, but anyways, this isn't a regression.